### PR TITLE
Add functionality to create a test package

### PR DESCRIPTION
**Pull request by adityapatwardhan**
## PR Summary

Add functionality to publish test package which has all the tests, test tools and other needed items in the expected folder structure. This will be published as an artifact for Windows and Linux daily builds.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by JamesWTruher — APPROVED** [review]
nothing blocking, and only 2 nits

File: build.psm1 [hunk 1]
```diff
@@ -2972,3 +2972,64 @@ $script:RESX_TEMPLATE = @'
...
+
+function New-TestPackage
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [string] $Destination
+    )
+
+    if ((Test-Path $Destination) -and -not (Test-Path $Destination -PathType Container))
```
[the remark this answers is not in the document]
Comment (vexx32) [reply build.psm1]: You mean -PathType Leaf?

[the remark this answers is not in the document]
Comment (adityapatwardhan) [reply build.psm1]: @vexx32 No I expect a container. I create the TestPackage.zip file in the directory provided.
@JamesWTruher if I combine the conditions I cannot know if it does not exist or it is not a directory.

[the remark this answers is not in the document]
Comment (vexx32) [reply build.psm1]: Given that this is a single `if` condition, I do not see a functional difference between these two:
```powershell
(Test-Path $Destination) -and -not (Test-Path $Destination -PathType Container)
# and
Test-Path $Destination -PathType Leaf
```
`-not ... -PathType Container` implies anything other than a container.... which is `-PathType Leaf`, isn't it?

Without breaking into more than one `if` statement, there is no functional difference, and indeed whether `$Destination` exists or not (provided it is either nonexistent or a container), you create a new item, suppressing the error if it already exists.

So... I see no reason to call `Test-Path` twice here. But then, it *is* later on Friday, so I might just be too weary for this at this point. 😄

[the remark this answers is not in the document]
Comment (adityapatwardhan) [reply build.psm1]: I now understand your comment. I will make the change.

**Pull request merged**